pages/api/writers/me.js

```diff
@@ -1,42 +1,31 @@
-import {CotterAccessToken} from "cotter-token-js";
 const {Pool} = require("pg");
 const connectionString = process.env.PG_CONNECTION_STRING;
 const pool = new Pool({connectionString});
+import { requireSession, users } from "@clerk/nextjs/api";
 
-export default async (req, res) => {
-    // Check that the authorization header exists
-    if (!("authorization" in req.headers)) {
-        res.statusCode = 401;
-        return res.end("Authorization header missing");
-    }
-
-    // Extract the token string
-    const auth = await req.headers.authorization;
-    const bearer = auth.split(" ");
-    const token = bearer[1];
-
+export default requireSession(async (req, res) => {
     try {
-        // Decode the Cotter JWT, "decoded.payload.identifier" is the user's email
-        const decoded = new CotterAccessToken(token);
-
-        if (req.method === "GET") {
-            // Get assignment by assignmentId
-            const query = `select writers.id,
-                                  writers.email,
-                                  requests.writer_at_max_requests
-                           from writers
-                           left join requests on requests.id = ANY (writers.requests)
-                           where writers.email like $1;`;
-            const {rows} = await pool.query(query, [decoded.payload.identifier]);
+      // Get user from Clerk API
+      const user = await users.getUser(req.session.userId);
+      // Get user
+      if (req.method === "GET") {
+          // Get assignment by assignmentId
+          const query = `select writers.id,
+                                writers.email,
```

Comment:
We're returning `email` here, so it should be `currentUser.email` in `belongsToCurrentUser()`